pkgs/applications/misc/flashprint/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv
+, lib
+, fetchurl
+, dpkg
+, glibc
+, gcc-unwrapped
+, autoPatchelfHook
+, libGLU
+, qtbase
+, wrapQtAppsHook
+}:
+
+stdenv.mkDerivation {
+  name = "flashprint";
+  version = "5.6.1";
+  src = fetchurl {
+    url = "https://en.fss.flashforge.com/10000/software/01e7742d5a721201cd33d4f8c083b63b.deb";
+    sha256 = "sha256-lvK4igv/L1zrqNtqbWtCyZrSZbkY8Ak/Li3NeVAH1YU=";
+  };
+
+  nativeBuildInputs = [ autoPatchelfHook dpkg wrapQtAppsHook ];
+  buildInputs = [ glibc gcc-unwrapped libGLU qtbase ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    dpkg -x $src $out
```

Comment:
If `dpkg` is in `nativeBuildInputs` it will automatically be unpacked to `./`, no need to do it manually.
So it would be enough to do `mv usr/* $out`

(Though you could override the default unwrapping if you have a custom `unpackPhase`)